pkg/admission/validator/shoot.go

```diff
@@ -17,18 +17,60 @@ package validator
 import (
 	"context"
 	"errors"
+	"fmt"
 	"reflect"
 
 	"github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws"
 	awsvalidation "github.com/gardener/gardener-extension-provider-aws/pkg/apis/aws/validation"
-
+	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
 	"github.com/gardener/gardener/pkg/apis/core"
 	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
 	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
 	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func (v *Shoot) validateShoot(ctx context.Context, shoot *core.Shoot) error {
+func NewShootValidator() extensionswebhook.Validator {
+	return &shoot{}
+}
+
+type shoot struct {
+	client  client.Client
+	decoder runtime.Decoder
+}
+
+// InjectScheme injects the given scheme into the validator.
+func (s *shoot) InjectScheme(scheme *runtime.Scheme) error {
+	s.decoder = serializer.NewCodecFactory(scheme).UniversalDecoder()
+	return nil
+}
+
+// InjectClient injects the given client into the validator.
+func (s *shoot) InjectClient(client client.Client) error {
+	s.client = client
+	return nil
+}
+
+func (s *shoot) Validate(ctx context.Context, new, old runtime.Object) error {
+	shoot, ok := new.(*core.Shoot)
+	if !ok {
+		return fmt.Errorf("wrong object type %T", new)
+	}
+
+	if old != nil {
+		oldShoot, ok := old.(*core.Shoot)
+		if !ok {
+			return fmt.Errorf("wrong object type %T for old object", old)
+		}
+		return s.validateShootUpdate(ctx, oldShoot, shoot)
+	}
+
+	return s.validateShootCreation(ctx, shoot)
+}
+
+func (v *shoot) validateShoot(ctx context.Context, shoot *core.Shoot) error {
```

Comment:
Inconsistent receiver names is a golint issue.

```suggestion
func (s *shoot) validateShoot(ctx context.Context, shoot *core.Shoot) error {
```